client/package.json

```diff
@@ -1,61 +1,61 @@
 {
-    "name": "client",
-    "version": "0.1.0",
-    "private": true,
-    "dependencies": {
-        "@react/collection-view": "file:libraries/react-collection-view-4.1.5.tgz",
-        "@react/react-spectrum": "file:libraries/react-react-spectrum-2.25.0.tgz",
-        "ace-builds": "^1.4.8",
-        "ajv": "^6.12.0",
-        "babel-polyfill": "^6.26.0",
-        "cross-os": "^1.3.0",
-        "form-data": "^3.0.0",
-        "js-yaml": "^3.13.1",
-        "react": "^16.13.0",
-        "react-ace": "^8.0.0",
-        "react-dom": "^16.13.0",
-        "react-scripts": "3.4.0",
-        "url": "^0.11.0"
-    },
-    "scripts": {
-        "start": "react-scripts start",
-        "build": "cross-os cross-os-build",
-        "eject": "react-scripts eject",
-        "pretest": "npm run build",
-        "preinstall": "npm i npm-force-resolutions --save-dev && npm-force-resolutions",
-        "test": "react-scripts test",
-        "posttest": "license-checker --summary",
-        "report-coverage": "nyc report --reporter=text-lcov > coverage.lcov"
-    },
-    "cross-os": {
-        "cross-os-build": {
-            "darwin": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
-            "linux": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
-            "win32": "npm i && react-scripts build && rm -rf ..\\server\\client-build && mv build ..\\server\\client-build"
-        }
-    },
-    "eslintConfig": {
-        "extends": "react-app"
-    },
-    "proxy": "http://localhost:9000",
-    "browserslist": {
-        "production": [
-            ">0.2%",
-            "not dead",
-            "not op_mini all"
-        ],
-        "development": [
-            "last 1 chrome version",
-            "last 1 firefox version",
-            "last 1 safari version"
-        ]
-    },
-    "devDependencies": {
-        "codecov": "^3.7.0",
-        "license-checker": "^25.0.1",
-        "npm-force-resolutions": "0.0.3"
-    },
-    "resolutions": {
-        "@react/collection-view": "file:libraries/react-collection-view-4.1.5.tgz"
-    }
+	"name": "client",
+	"version": "0.1.0",
+	"private": true,
+	"dependencies": {
+		"@react/collection-view": "file:libraries/react-collection-view-4.1.5.tgz",
+		"@react/react-spectrum": "file:libraries/react-react-spectrum-2.25.0.tgz",
+		"ace-builds": "^1.4.8",
+		"ajv": "^6.12.0",
+		"babel-polyfill": "^6.26.0",
+		"cross-os": "^1.3.0",
+		"form-data": "^3.0.0",
+		"js-yaml": "^3.13.1",
+		"react": "^16.13.0",
+		"react-ace": "^8.0.0",
+		"react-dom": "^16.13.0",
+		"react-scripts": "3.4.0",
+		"url": "^0.11.0"
+	},
+	"scripts": {
+		"start": "PORT=3000 react-scripts start",
```

Comment:
No, we don't want to hardcode this. What if port 3000 is taken? There is already a method built into react-scripts to find an open port

Reply:
So this only is written in dev mode because the original code did not look for a new port, client failed if 3000 was still in use, this is why I even bothered to fix the port issues. 

I think we should run set-port.js no matter what so that client has an available port to listen on.
Else I will revert all the port changes back and make a note if client port is already in use, kill node processes then re-start the server - it might not even matter after the stop() is implemented.

Reply:
just to clarify, the client only listens to a port in dev mode. So this is only relevant for dev mode.

react scripts has built-in functionality to find open ports if the default one is already taken. See here what happens when I try to run `npm start` in client when port 3000 is already taken:
<img width="603" alt="Screen Shot 2020-06-16 at 12 12 32 PM" src="https://user-images.githubusercontent.com/20048485/84818093-c7e2b700-afca-11ea-8489-990f4c75603c.png">


So as far as I was concerned, there was no reason to add any other functionality to change the client port. Maybe with the concurrent processes in server, this react feature gets lost

Reply:
I was just able to clarify that using concurrent processes ruins the ootb new port feature in react:
<img width="782" alt="Screen Shot 2020-06-16 at 12 16 50 PM" src="https://user-images.githubusercontent.com/20048485/84818438-58b99280-afcb-11ea-85a3-bda0ce9a16e9.png">

Reply:
Ok in working on this project, I never got the first message to change the port, I always got the second message.

For the comment about the concurrent processes, is this related to using `concurrently` npm package in the package.json srcipts section? If so, would we want to remove all the port changes and separate what nodemon is monitoring?